plugins/org.obeonetwork.m2doc/src/org/obeonetwork/m2doc/generator/TemplateProcessor.java

```diff
@@ -848,7 +861,7 @@ private void setGenericParameters(AbstractProviderClient templateProvider, Map<S
         EMap<String, Object> optionsMap = templateProvider.getOptionValueMap();
         Set<Entry<String, Object>> optionsMapEntries = optionsMap.entrySet();
         for (Entry<String, Object> optionsMapEntry : optionsMapEntries) {
-            if (optionTypes == null || optionTypes.get(optionsMapEntry.getKey()) == null) {
+            if ((optionTypes != null && optionTypes.get(optionsMapEntry.getKey()) == null) || optionTypes == null) {
```

Comment:
Why do you change that ? It's strictly equivalent but more complex after your change!
(FYI, I changed that when I worked on wtable I believe)